presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -797,10 +799,15 @@ public Map<String, Optional<Partition>> getPartitionsByNames(MetastoreContext me
 
     private Optional<Partition> loadPartitionByName(KeyAndContext<HivePartitionName> partitionName)
     {
+        String databaseName = partitionName.getKey().getHiveTableName().getDatabaseName();
+        String tableName = partitionName.getKey().getHiveTableName().getTableName();
+        Optional<Table> table = getTable(partitionName.getContext(), databaseName, tableName);
```

Comment:
this will result an additional getTable roundtrip call to metastore ? this might cause performance regression

Reply:
this should get it from the cache in case it is fetched already. So I guess there should not be any performance regression. @NikhilCollooru

Reply:
But Table cache is not enabled by default right. So if we dont have cache enabled it would result in an additional call ?

Reply:
yes, table cache is not enabled by default. But we actually need table object in GlueToPrestoConverter constructor call to be able to get serde object and fix the original issue reported in https://github.com/prestodb/presto/issues/15869. I cannot think of any way to get the serde without getting the table object. 
I am open hear your suggestions around the same. @NikhilCollooru

Reply:
I am wondering if you can gate the code changes in this PR behind a config. So that we enable this code path only when one needs to support accessing tables using Glue.

Reply:
The code changes in this PR involve API signature changes in ExtendedHiveMetastore and are present across multiple files. Also the cache structure got modified in CachingHiveMetastore. Hence guarding these changes behind a config seems a tedious task and will make the code too dirty. 
Also I am not sure how many users are actually using Presto without enabling cache. Most users will tend to use Presto with Caching enabled I believe. One workaround I can think of is enabling cache by default with some sane values for expiry and max size. I guess this should not be a breaking change for users, since Presto already has the concept of open and closed partitions. 
Again I am open to hearing your thoughts on this @NikhilCollooru

Reply:
After reading code i see that we memoize/cache all calls to Metastore  within a transaction. Meaning getTable on a particular table will go to Metastore only once and duplicate getTable future calls on that same table will be served from the cache. This is irrespective of whether table cache is enabled or not.
https://github.com/prestodb/presto/blob/master/presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java#L160

@pratyakshsharma can you run some tests and confirm that the actual number of getTable calls to Metastore for a given query (not involving glue tables) will remain same before and after this PR ?

Reply:
Sure, will get back to you on this.

Reply:
As I checked, it looks like SemiTransactionalHiveMetastore has issues with creating CachingHiveMetastore instance properly. With metastore types other than `alluxio`, CachingHiveMetastore has another CachingHiveMetastore as delegate, which results in routing the calls again to CachingHiveMetastore. Due to this, 2 extra getTable calls were being made to metastore. Here is the plan to fix this - 
1. Fix creation of CachingHiveMetastore instance. (this removes 1 extra call)
2. Reintroduce api getPartitionsByNames which takes databaseName and tableName as parameters, and call it in case of thrift hive metastore. (this removes 1 extra call). 

I will be taking care of this fix in my next commit @NikhilCollooru

Reply:
@NikhilCollooru please take a pass. I can confirm the number of getTable calls to metastore do not increase after my changes. In fact, one redundant getTable call to CachingHiveMetastore is now reduced.